src/main/java/com/auth0/json/mgmt/logstreams/LogStream.java

```diff
@@ -0,0 +1,131 @@
+package com.auth0.json.mgmt.logstreams;
+
+import com.fasterxml.jackson.annotation.*;
+
+import java.util.Map;
+
+/**
+ * Represents an Auth0 Log Stream object. Related to the {@linkplain com.auth0.client.mgmt.LogStreamsEntity}
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class LogStream {
+
+    @JsonProperty("id")
+    private String id;
+    @JsonProperty("name")
+    private String name;
+    @JsonProperty("type")
+    private String type;
+    @JsonProperty("status")
+    private String status;
+    @JsonProperty("sink")
+    private Map<String, Object> sink;
+
+    /**
+     * Creates a new LogStream instance and sets the type.
+     *
+     * @param type The type of the Log Stream.
+     */
+    @JsonCreator
+    public LogStream(@JsonProperty("type") String type) {
+        this.type = type;
+    }
+
+    /**
+     * Creates a new LogStream instance.
+     */
+    public LogStream() {
+        this(null);
+    }
+
+    /**
+     * Get the ID of this Log Stream.
+     *
+     * @return The ID of this Log Stream.
+     */
+    @JsonProperty("id")
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Get the name of this Log Stream.
+     *
+     * @return The name of this Log Stream.
+     */
+    @JsonProperty("name")
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Sets the name of this Log Stream.
+     *
+     * @param name the name to set.
+     */
+    @JsonProperty("name")
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Gets the type of this Log Stream.
+     *
+     * @return The type of this Log Stream.
+     */
+    @JsonProperty("type")
+    public String getType() {
+        return type;
+    }
+
+    /**
+     * Sets the type of this Log Stream.
+     *
+     * @param type The type to set.
+     */
+    @JsonProperty("type")
+    public void setType(String type) {
```

Comment:
if this cannot be changed once set, I'd leave the option to set it only through the constructor and remove this setter. That's the pattern we follow on other places, afaik.

Reply:
I'm a bit concerned about the usability of such a pattern for this API. If you even send the `type` field on an update request, it will fail. Since we're using the same model for create/update/get, that means to update a LogStream you'd have to create it by passing `null` to the constructor to create a new instance that can work for the update request. Is that the pattern followed in other places, and if so, is it drastically inconsistent to keep a default constructor to avoid that scenario?